data/data-pipeline/data_pipeline/etl/sources/cdc_life_expectancy/etl.py

```diff
@@ -23,32 +35,122 @@ def __init__(self):
-            file_url=self.FILE_URL,
-            download_file_name=download_file_name,
+            file_url=self.USA_FILE_URL,
+            download_file_name=all_usa_download_file_name,
+            verify=True,
+        )
+
+        pandas_read_csv_dtype_settings = {
+            # The following need to remain as strings for all of their digits, not get converted to numbers.
+            self.TRACT_INPUT_COLUMN_NAME: "string",
+            self.STATE_INPUT_COLUMN_NAME: "string",
+        }
+
+        all_usa_raw_df = pd.read_csv(
+            filepath_or_buffer=all_usa_download_file_name,
+            dtype=pandas_read_csv_dtype_settings,
+            low_memory=False,
+        )
+
+        # Check which states are missing
+        states_in_life_expectancy_usa_file = all_usa_raw_df[
+            self.STATE_INPUT_COLUMN_NAME
+        ].unique()
+
+        # Find which states are missing from the expected set.
+        states_missing = sorted(
+            list(
+                self.EXPECTED_STATES_SET
+                - set(states_in_life_expectancy_usa_file)
+            )
+        )
+
+        if states_missing != self.EXPECTED_MISSING_STATES:
+            raise ValueError(
+                "LEEP data has changed. The states missing from the data are "
+                "no longer the same."
+            )
+
+        logger.info("Downloading data for Maine")
+        maine_download_file_name = (
+            self.get_tmp_path() / "cdc_life_expectancy" / "maine.csv"
+        )
+        download_file_from_url(
+            file_url=self.MAINE_FILE_URL,
+            download_file_name=maine_download_file_name,
             verify=True,
         )
+        maine_raw_df = pd.read_csv(
+            filepath_or_buffer=maine_download_file_name,
+            dtype=pandas_read_csv_dtype_settings,
+            low_memory=False,
+        )
 
-        self.raw_df = pd.read_csv(
-            filepath_or_buffer=download_file_name,
-            dtype={
-                # The following need to remain as strings for all of their digits, not get converted to numbers.
-                self.TRACT_INPUT_COLUMN_NAME: "string",
-            },
+        logger.info("Downloading data for Wisconsin")
+        wisconsin_download_file_name = (
+            self.get_tmp_path() / "cdc_life_expectancy" / "wisconsin.csv"
+        )
+        download_file_from_url(
+            file_url=self.WISCONSIN_FILE_URL,
+            download_file_name=wisconsin_download_file_name,
+            verify=True,
+        )
+        wisconsin_raw_df = pd.read_csv(
+            filepath_or_buffer=wisconsin_download_file_name,
+            dtype=pandas_read_csv_dtype_settings,
             low_memory=False,
         )
 
+        combined_df = pd.concat(
+            objs=[all_usa_raw_df, maine_raw_df, wisconsin_raw_df],
+            ignore_index=True,
+            verify_integrity=True,
+            axis=0,
+        )
+
+        states_in_combined_df = combined_df[
```

Comment:
again this is imo a slightly confusing way to do this! 

I also wonder -- should we move something like this into a util and validate all of the data we expect to be nationally comprehensive with this, rather than catch these errors kind of one off?